app/views/shared/_sidepanel.html.erb

```diff
@@ -0,0 +1,7 @@
+<% if user_signed_in? %>
+<div class="w3-sidebar w3-bar-block " style="width:6%">
+    <li><button class="btn btn-light btn-block"><%=link_to ('<i class="fas fa-2x fa-home"></i>').html_safe, root_path %></button></li>
```

Comment:
don't use html_safe, use sanitize function there
https://apidock.com/rails/ActionView/Helpers/SanitizeHelper/sanitize
Please read why we should not use html_safe and use sanitize

Reply:
done